cmd/client/command/common.go

```diff
@@ -182,10 +186,20 @@ func readFromFileOrStdin(specFile string, cmd *cobra.Command) ([]byte, string) {
 		Kind string `yaml:"kind"`
 		Name string `yaml:"name"`
 	}
-	err = yaml.Unmarshal(buff, &spec)
-	if err != nil {
-		ExitWithErrorf("%s failed, invalid spec: %v", cmd.Short, err)
-	}
 
-	return buff, spec.Name
+	yamlDocs := strings.Split(string(buff), yamlSeparator)
+
+	// make sure each yaml doc valid
+	for _, yamlDoc := range yamlDocs {
+		if len(strings.TrimSpace(yamlDoc)) == 0 {
+			continue
+		}
+		err = yaml.Unmarshal([]byte(yamlDoc), &spec)
+		if err != nil {
+			ExitWithErrorf("%s failed, invalid spec: %v", cmd.Short, err)
+			break
+		}
+
+		handler(yamlDoc, spec.Name)
+	}
```

Comment:
propose to validate all `ymalDoc`s before call `handler` to any of them, this makes the whole process more like a transaction (though it is not).